p2p/simulations/network.go

```diff
@@ -155,6 +156,12 @@ type ConnEvent struct {
 	conn   *Conn
 }
 
+type MsgEvent struct {
+	Action string
+	Type   string
+	msg    *Msg
```

Comment:
true but the others don't have json tags for their unexported fields either. It doesn't need be consistent? Do we need to serialize it?